crates/api_common/src/lib.rs

```diff
@@ -580,10 +579,9 @@ pub async fn remove_user_data_in_community(
   Ok(())
 }
 
-pub fn check_image_has_local_domain(url: &Option<String>) -> Result<(), LemmyError> {
+pub fn check_image_has_local_domain(url: &Option<DbUrl>) -> Result<(), LemmyError> {
```

Comment:
I changed this because I found that locally: 

- I tried to update the site, but it found the site icon was the wrong local domain and errored.
- So tried to delete the site image, which goes to `Some("")`, which the diesel_overwrite_image function checks for, and transforms into `Some(None))` IE a database set to null. This failed because this domain check was done before that overwrite function.
- To fix I moved this check to after the overwrite function.